File: python/socketd/socketd/transport/stream/impl/SendStreamImpl.py
from typing import Callable

from socketd.transport.core.Costants import Constants
from socketd.transport.stream.SendStream import SendStream
from socketd.transport.stream.impl.StreamBase import StreamBase


class SendStreamImpl(StreamBase, SendStream):
    def __init__(self, sid: str):
        super().__init__(sid, Constants.DEMANDS_ZERO, 0)

    def is_done(self):
        return True

    async def on_reply(self, reply):
        ...

    def then_error(self, onError: Callable[[Exception], None]) -> 'SendStream':
        super()._then_error_do(onError)
        return self


    def then_progress(self, onProgress: Callable[[bool, int, int], None]) -> 'SendStream':
        super().then_progress(onProgress)
        return self

File: javascript/socketd/src/socketd/transport/stream/impl/SendStreamImpl.ts
import {Constants} from "../../core/Constants";
import {MessageInternal} from "../../core/Message";
import {SendStream} from "../Stream";
import {StreamBase} from "./StreamBase";

export class SendStreamImpl extends StreamBase<SendStream> implements SendStream{
    constructor(sid: string) {
        super(sid, Constants.DEMANDS_ZERO, 0);
    }

    isDone(): boolean {
        return true;
    }

    onReply(reply: MessageInternal) {
    }
}